sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -149,10 +153,17 @@ case class Like(left: Expression, right: Expression) extends StringRegexExpressi
     } else {
       val pattern = ctx.freshName("pattern")
       val rightStr = ctx.freshName("rightStr")
+      val escapeChar = escapeCharOpt.map { str =>
+        if (str.equals("\"") || str.equals("\\")) {
```

Comment:
Is this to avoid double escape in codegen? Can you leave some comments?

Reply:
I remember this is to accept [the query](https://github.com/apache/spark/pull/25001#issuecomment-511803526). The suggested looks nice and we need comments here.

Reply:
@HyukjinKwon @maropu I added the comments for explain the double escape.